py3status/modules/wwan_status_nm.py

```diff
@@ -0,0 +1,252 @@
+                    if status['state'] == 11:  # connected
+                        # Get network config
+                        bearer = modem_proxy.Bearers[0]
+                        network_config = self._get_network_config(bearer)
+
+                        # Add network config to data dict
+                        data.update(network_config)
+
+                        if data['ip']:
+                            if self.consider_3G_degraded and data['netgen'] != 'LTE':
+                                response['color'] = self.py2.COLOR_DEGRADED
+                            else:
+                                response['color'] = self.py3.COLOR_GOOD
+                            response['full_text'] = self.py3.safe_format(
+                                self.format_up, data)
+                        else:
+                            response['color'] = self.py3.COLOR_DEGRADED
+                            response['full_text'] = self.py3.safe_format(
+                                self.format_no_service, data)
+
+                    else:  # disconnected
+                        response['full_text'] = self.py3.safe_format(
+                            self.format_down, data)
+                        response['color'] = self.py3.COLOR_BAD
+
+                except:
+                    data['error'] = self.states[status['state']]
+                    response['color'] = self.py3.COLOR_BAD
+                    response['full_text'] = self.py3.safe_format(self.format_error, data)
+
+        return response
+
+    # get network config function
+    def _get_network_config(self, bearer):
+        try:
+            bearer_proxy = self.bus.get(STRING_MODEMMANAGER_DBUS, bearer)
+
+            network_config = {}
+
+            # Get ipv4 config
+            ipv4 = bearer_proxy.Ip4Config
+            if ipv4['address']:
+                network_config['ipv4_address'] = ipv4['address']
+                network_config['ipv4_dns1'] = ipv4['dns1']
+                network_config['ipv4_dns2'] = ipv4['dns2']
+                network_config['ip'] = ipv4['address']
+            else:
+                network_config['ipv4_address'] = ''
+                network_config['ipv4_dns1'] = ''
+                network_config['ipv4_dns2'] = ''
+                network_config['ip'] = STRING_NO_IP
+
+            # Get ipv6 network config
+            ipv6 = bearer_proxy.Ip6Config
+            if ipv6['address']:
+                network_config['ipv6_address'] = ipv6['address']
+                network_config['ipv6_dns1'] = ipv6['dns1']
+                network_config['ipv6_dns2'] = ipv6['dns2']
+                network_config['ip6'] = ipv6['address']
+            else:
+                network_config['ipv6_address'] = ''
+                network_config['ipv6_dns1'] = ''
+                network_config['ipv6_dns2'] = ''
+                network_config['ip6'] = STRING_NO_IP
+
+        except:
+            pass
+
+        finally:
+            return network_config
```

Comment:
Not understanding this... but this, I suppose, could fail with `bearer_proxy = self.bus.get(STRING_MODEMMANAGER_DBUS, bearer)` leaving us with nonexistent `network_config` so this dict should be up a bit more... to be more safe.